# Mini-Project 2: Solar Micro-Grid Dispatch Planner

This project plans solar and battery usage for a rural health centre
in Kasese using two demand equations:

- 3x + 2y = D1
- 4x + y = D2

Here, x is solar usage and y is battery usage, measured in kWh.

In [3]:
from pathlib import Path
import sys

# Add the project's src folder to Python's import path.
project_folder = Path.cwd().parent
sys.path.insert(0, str(project_folder / "src"))

# Import from microgrid.py and create the health centre object.
from microgrid import MicroGrid

kasese_health_centre = MicroGrid("Kasese Health Centre")

# Display the matrix checks before solving the demand equations.
print("Health centre:", kasese_health_centre.centre_name)
print("Determinant:", round(kasese_health_centre.determinant(), 2))
print("Condition number:", round(kasese_health_centre.condition_number(), 2))

Health centre: Kasese Health Centre
Determinant: -5.0
Condition number: 5.83


### Checking the demand equations

The coefficient matrix has a determinant of −5. Since this is non-zero,
the system has a unique solution for each pair of demand values.

The condition number is approximately 5.83, indicating moderate sensitivity
to changes in demand. It is not close to singular. However, a unique
mathematical solution may still contain negative solar or battery usage,
which would be physically infeasible.